corehq/apps/userreports/views.py

```diff
@@ -954,10 +954,16 @@ def post(self, request, *args, **kwargs):
                 messages.success(request, _("Data source created for '{}'".format(app_source.source)))
             else:
                 assert app_source.source_type == 'form'
-                xform = Form.get_form(app_source.source)
-                data_source = get_form_data_source(app, xform)
+                found_form = None
+                for f in app.get_forms():
+                    if f.unique_id == app_source.source:
+                        found_form = f
+                if found_form is None:
```

Comment:
could use `app.get_form(app_source.source)` which raises a `FormNotFoundException` if no form found.

Reply:
thanks, not sure how I missed that. updated